### ISSUE TITLE
Add sessionId to CounterRequestContext

### ISSUE THREAD

**Issue by kortov** [javamelody/javamelody#872]
Hello, we've stumbled upon a problem to bind requests returned from "ExternalAPI" to sessions. (At least our Jira doesn't serialize sessions, so we can't get a remote user from a request). It would be nice to get sessionId in part=currentRequests.

 I've already added sessionId locally in the code and it's working but I'd like to Pull Request it if you don't mind. I decided to create an issue first.

I'm not sure if it needed to include sessionId to Current requests html report, I haven't added it to the code yet.

**Comment by evernat** [comment]
Yes, I am ok to add a sessionId attribute in CounterRequestContext. You can create a PR.
We could add a link to session details from the current requests in the the html report, but in my opinion at the moment it would not give an important information.

By the way, I see that the MonitoringFilter [calls](https://github.com/javamelody/javamelody/blob/master/javamelody-core/src/main/java/net/bull/javamelody/MonitoringFilter.java#L234) `httpRequest.getRemoteUser()` to create the CounterRequestContext. It could do better to find the remote user, for example:
```java
String remoteUser = httpRequest.getRemoteUser();
if (remoteUser == null) {
   final HttpSession session = httpRequest.getSession(false);
   if (session != null) {
      remoteUser = session.getAttribute(SessionListener.SESSION_REMOTE_USER);
   }
}
```
In many JIRA versions, it would find the remote user (name of the connecter user) which is not found currently for the http CounterRequestContext.

**Comment by kortov** [comment]
I created a PR

I didn't provide new tests (because I added private methods and it's quite tricky to test them) but of course it's better to have a testable code, so I try to add tests if you want to.

And I guess it would be nice to have tests where bindContext is invoked with not nullable httpRequest , but I didn't elaborate on tests because now it's hard to run tests in IDEA (there are some java version and dependency errors when I run test normally through green arrow). Do you can run tests for the project locally?

And I hoped that you have some CI on PR's but there are no of it. Do you have plans on some travis CI? May be create an issue and try to add it? :)

**Comment by kortov** [comment]
I guess I didn't PR properly, it doesn't show diffs only replaced files, I'll try to fix it

**Comment by evernat** [comment]
Since you are using the external API, you may be interested to know that the current master branch (and the next release) includes the availability of the external API as JSON format using Google Gson (format=gson), as an alternative to XStream (format=json or format=xml).
The JSON results with Gson are a bit different than with XStream.
See https://github.com/javamelody/javamelody/wiki/ExternalAPI#gson

**Comment by kortov** [comment]
Thank you for info. When we tried to parse results from monitoring?part=currentRequests&format=json we often stumbled upon an issue (if you'd like I can open an issue) that json was with errors like "SyntaxError: JSON.parse: unexpected end of data at line 13452 column 28 of the JSON data" in Mozilla browser. Unfortunately I can't provide you with a full json and I can't debug the plugin on the production JIRA but in mozilla and chrome the json cuts off  on parentContext word 

``` ],
      "jcacheInformationsList": [],
      "jobInformationsList": [],
      "hsErrPidList": [],
      "webXmlExists": true,
      "pomXmlExists": false,
      "springBeanExists": false
    },
    [
      {
        "parentCounter": {
          "displayed": true,
          "name": "http",
          "errorCounter": false,
          "storageName": "http",
          "iconName": "dbweb.png",
          "childCounterName": "sql",
          "requests": [],
          "rootCurrentContextsByThreadId": [],
          "startDate": "2019-12-25 07:23:51.989 UTC",
          "maxRequestsCount": 10000,
          "estimatedMemorySize": 0
        },
        "requestName": "/rest/gadget/1.0/issueTable/filter ajax GET",
        "completeRequestName": "/rest/gadget/1.0/issueTable/filter?num=10&tableContext=jira.table.cols.dashboard&addDefault=false&columnNames=issuetype&columnNames=issuekey&columnNames=summary&columnNames=customfield_10405&columnNames=duedate&columnNames=priority&columnNames=status&enableSorting=true&paging=true&showActions=true&filterId=72128&sortBy=&startIndex=0&_=1577258631403 ajax GET",
        "remoteUser": "KosSM",
        "threadId": 397,
        "startTime": 1577258616591,
        "startCpuTime": 1314355225300,
        "startAllocatedBytes": 155305847792,
        "sessionId": "FE998404A76E69E9FFA4BBD67FD3DAB8",
        "childHits": 0,
        "childDurationsSum": 0
      },
      {
        "parentCounter": {
          "displayed": true,
          "name": "http",
          "errorCounter": false,
          "storageName": "http",
          "iconName": "dbweb.png",
          "childCounterName": "sql",
          "requests": [],
          "rootCurrentContextsByThreadId": [],
          "startDate": "2019-12-25 07:23:51.989 UTC",
          "maxRequestsCount": 10000,
          "estimatedMemorySize": 0
        },
        "currentChildContext": {
          "parentCounter": {
            "displayed": true,
            "name": "jsp",
            "errorCounter": false,
            "storageName": "jsp",
            "iconName": "jsp.png",
            "childCounterName": "sql",
            "requests": [],
            "rootCurrentContextsByThreadId": [],
            "startDate": "2019-12-25 07:23:51.989 UTC",
            "maxRequestsCount": 10000,
            "estimatedMemorySize": 0
          },
          "parentContext": 
```

so we switched to an xml format because it has no that problem. We're using now plugin from master with this sessionId feature, I guess we'll update a plugin only after you release a new version

**Comment by evernat** [comment]
I have reproduced your last issue. See #884

**Comment by kortov** [comment]
Ok, thanks
